services/users/users.go

```diff
@@ -5,29 +5,77 @@ package users
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"runtime"
+	"sync"
 
+	"github.com/mattermost/mattermost-server/v5/einterfaces"
 	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/services/cache"
 	"github.com/mattermost/mattermost-server/v5/shared/i18n"
 	"github.com/mattermost/mattermost-server/v5/shared/mlog"
 	"github.com/mattermost/mattermost-server/v5/store"
 )
 
 type UserService struct {
-	store  store.UserStore
-	config func() *model.Config
+	store        store.UserStore
+	sessionStore store.SessionStore
+	sessionCache cache.Cache
+	sessionPool  sync.Pool
+	metrics      einterfaces.MetricsInterface
+	cluster      einterfaces.ClusterInterface
+	config       func() *model.Config
 }
 
 type UserCreateOptions struct {
 	Guest      bool
 	FromImport bool
 }
 
-func New(s store.UserStore, cfgFn func() *model.Config) *UserService {
-	return &UserService{
-		store:  s,
-		config: cfgFn,
+// ServiceInitializer is used to initialize the UserService.
+type ServiceInitializer struct {
```

Comment:
Maybe `ServiceConfig`? 0/5

Reply:
Sure, I'll include that in the next PR 👍